lib/expressapp.js

```diff
@@ -369,13 +369,22 @@ ExpressApp.prototype.start = function(opts, cb) {
   router.get('/v1/addresses/', function(req, res) {
     getServerWithAuth(req, res, function(server) {
       var opts = {};
-      if (req.query.limit) opts.limit = +req.query.limit;
-      opts.reverse = (req.query.reverse == '1');
-
-      server.getMainAddresses(opts, function(err, addresses) {
-        if (err) return returnError(err, res, req);
-        res.json(addresses);
-      });
+      if (req.query.walletId && req.query.address) {
```

Comment:
Querying a single address should have its own endpoint with a route like this: `/v1/addresses/:address/`